common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -3468,6 +3468,10 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "          (Use with property hive.server2.custom.authentication.class)\n" +
         "  PAM: Pluggable authentication module\n" +
         "  NOSASL:  Raw transport"),
+    HIVE_SERVER2_TRUST_DOMAIN("hive.server2.trust.domain", "",
+            "Specifies the host or a domain to trust connections from. Authentication is skipped " +
+                    "for any connection coming from this domain or the host. By default it is " +
+                    "empty, which means that all the connections to HiveServer2 are authenticated."),
```

Comment:
Explaining the logic that this property provides a suffix (for endswith check) rather than requiring an exact host name, could help.

Reply:
Hmm. I have rephrased the explanation, (also incorporating the request from @prasanthj. Please check if this conveys the right meaning.

Reply:
Done.